src/components/Nav.tsx

```diff
@@ -2,18 +2,23 @@ import { Link } from 'react-router-dom';
 import LinkBorder from './LinkBorder';
 import { useState, useEffect } from 'react';
 import { AnimatePresence, motion } from 'framer-motion';
+import { debounce } from 'debounce';
 
 const Nav = () => {
   const [clicked, setClicked] = useState(false);
   const [width, setWidth] = useState(window.innerWidth);
 
   useEffect(() => {
-    const handleWindowResize = () => {
+    const handleWindowResize = debounce(() => {
       setWidth(window.innerWidth);
       console.log(width);
-    };
+    }, 300);
```

Comment:
Inace nebi treba rjesavat responzivnost javascriptom nego u css-u, al posto ja ne znam tailwind ne znam kako se s njime to radi. Spremanje widtha u state je bas skolski primjer za koristit debounce, jer inace ce ti se komponenta rerenderat na jako puno puta kad resizeaš :)